### PR TITLE
Improve Postgis raster support for libpq services (fixes #3820)

### LINKED ISSUE #3820: Postgis raster driver does not work with services
Postgis vector driver works well with service files. The same is not true for Postgis raster driver.

## Expected behavior and actual behavior.

Running:
```bash
gdalinfo "PG:service=ortos schema=ortos table=desertas mode=2"
```
fails with:
```
You must specify at least a db name
```

`gdalinfo` works if we use the dbname in the connection string, as in:
```
gdalinfo "PG:service=ortos dbname=ortos schema=ortos table=desertas mode=2"
```
The usage of the `dbname` here is redundant, since it is already present in the service file.

This is almost the same issue as #2373 (for the vetor driver), that was already fixed by @rouault 

## Steps to reproduce the problem.

1. Create a service file, like:
```
[ortos]
host=irig-madeira.pt
port=5432
dbname=ortos
user=drote
password=xxxx
```
2. Add rasters to the database
3. Access rasters with raster tools, like `gdalinfo`or `gdal_translate` using `service=ortos` in the connection string.

## Operating system

Ubuntu 20.04

## GDAL version and provenance

Current master

## PR

Under preparation

### PR DESCRIPTION
**Pull request by jgrocha**
## What does this PR do?

This PR allow users to use just the service name in Postgis raster driver connection strings.

All the support was already there. The condition to check if there is a dbname was changed, to check if we have a database or a service name. We only report failure if both are not present.

## Examples

```bash
$ gdalinfo "PG:service=ortos schema=ortos table=ortos_2018_2k_desertas column=rast mode=2"
Driver: PostGISRaster/PostGIS Raster driver
Files: none associated
Size is 96000, 220000
Coordinate System is:
BOUNDCRS[
    SOURCECRS[
        PROJCRS["PTRA08 / UTM zone 28N",
(...)
Data axis to CRS axis mapping: 1,2
Origin = (354000.000000000000000,3607300.000000000000000)
Pixel Size = (0.100000000000000,-0.100000000000000)
Corner Coordinates:
Upper Left  (  354000.000, 3607300.000) ( 16d33'20.63"W, 32d35'36.87"N)
Lower Left  (  354000.000, 3585300.000) ( 16d33' 8.34"W, 32d23'42.66"N)
Upper Right (  363600.000, 3607300.000) ( 16d27'12.47"W, 32d35'41.28"N)
Lower Right (  363600.000, 3585300.000) ( 16d27' 1.00"W, 32d23'47.04"N)
Center      (  358800.000, 3596300.000) ( 16d30'10.60"W, 32d29'42.00"N)
Band 1 Block=2048x2048 Type=Byte, ColorInterp=Undefined
  Overviews: 12000x27500, 6000x13750, 3000x6875, 1500x3438, 750x1719, 375x859, 188x430
Band 2 Block=2048x2048 Type=Byte, ColorInterp=Undefined
  Overviews: 12000x27500, 6000x13750, 3000x6875, 1500x3438, 750x1719, 375x859, 188x430
Band 3 Block=2048x2048 Type=Byte, ColorInterp=Undefined
  Overviews: 12000x27500, 6000x13750, 3000x6875, 1500x3438, 750x1719, 375x859, 188x430
Band 4 Block=2048x2048 Type=Byte, ColorInterp=Undefined
  Overviews: 12000x27500, 6000x13750, 3000x6875, 1500x3438, 750x1719, 375x859, 188x430
```

```bash
$ gdalinfo "PG:schema=ortos table=ortos_2018_2k_desertas column=rast mode=2"
ERROR 1: You must specify at least a db name or a service name
gdalinfo failed - unable to open 'PG:schema=ortos table=ortos_2018_2k_desertas column=rast mode=2'.
```
## Environment

* OS: Ubuntu 20.04

### CODE REVIEW

File: gdal/frmts/postgisraster/postgisrasterdriver.cpp [hunk 1]
```diff
@@ -68,21 +68,11 @@ PostGISRasterDriver::~PostGISRasterDriver() {
...
-    if( pszPortIn == nullptr ) pszPortIn = "(null)";
-    if( pszUserIn == nullptr ) pszUserIn = "(null)";
-    CPLString osKey = pszDbnameIn;
-    osKey += "-";
-    osKey += pszHostIn;
-    osKey += "-";
-    osKey += pszPortIn;
-    osKey += "-";
-    osKey += pszUserIn;
+    CPLString osKey = pszConnectionString;
```
Comment (rouault) [inline gdal/frmts/postgisraster/postgisrasterdriver.cpp:75]: this change is a bit too extreme. The connection string also contains element related to tables that do not determine the PG connection itself. You'd rather want to add an extra parameter for the service name

Comment (jgrocha) [reply gdal/frmts/postgisraster/postgisrasterdriver.cpp]: Agree :-)

The key for the internal map is now like this: `service-(null)-(null)-(null)-123456789`

File: gdal/frmts/postgisraster/postgisrasterdataset.cpp [hunk 2]
```diff
@@ -3021,13 +3026,14 @@ GetConnection(const char * pszFilename, char ** ppszConnectionString,
     OutDBResolution* peOutDBResolution)
 {
     PGconn * poConn = nullptr;
+    char * pszService = nullptr;
```
Comment (rouault) [inline gdal/frmts/postgisraster/postgisrasterdataset.cpp:3029]: should be CPFree 'd() at the end of the function

File: gdal/frmts/postgisraster/postgisrasterdataset.cpp [hunk 3]
```diff
@@ -2786,24 +2787,26 @@ GetConnectionInfo(const char * pszFilename,
...
+                "You must specify at least a db name or a service name");
 
         CSLDestroy(papszParams);
 
         return false;
-    }
-
-    *ppszDbname =
-        CPLStrdup(CPLParseNameValue(papszParams[nPos], nullptr));
+    }
```
Comment (rouault) [inline gdal/frmts/postgisraster/postgisrasterdataset.cpp:2803]: extra white space added at that line and following one

File: gdal/frmts/postgisraster/postgisrasterdataset.cpp (hunk 2 above)
Comment (jgrocha) [reply gdal/frmts/postgisraster/postgisrasterdataset.cpp:3029]: Sure. Sorry for that.

File: gdal/frmts/postgisraster/postgisrasterdataset.cpp (hunk 3 above)
Comment (jgrocha) [reply gdal/frmts/postgisraster/postgisrasterdataset.cpp]: Fixed.

**Pull request merged**